api/src/db.js

```diff
@@ -163,4 +164,37 @@ if (UNIT_TEST_ENV) {
 
     throw new Error(`Error while saving docs: ${errors.join(', ')}`);
   };
+
+  const DEFAULT_SECURITY_STRUCTURE = {
+    names: [],
+    roles: [],
+  };
+
+  module.exports.addRoleToSecurity = async (dbname, role, addAsAdmin) => {
+    if (!dbname || !role) {
+      return;
+    }
+
+    const securityUrl = new URL(environment.serverUrl);
+    securityUrl.pathname = `${dbname}/_security`;
+
+    const securityObject = await rpn.get({ url: securityUrl.toString(), json: true });
+    const property = addAsAdmin ? 'admins' : 'members';
+    if (!securityObject[property]) {
+      securityObject[property] = DEFAULT_SECURITY_STRUCTURE;
+    }
+
+    if (!securityObject[property].roles) {
```

Comment:
This is weird right, because we've found the security object in the db and now we're throwing, which will be a bad time if this is run during migration. Can't we just add the `roles` array to proactively "fix" the security object?